dali/util/uri.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_UTIL_URI_H_
+#define DALI_UTIL_URI_H_
+
+#include <cstring>
+#include <string>
+
+namespace dali {
+
+struct ParsedURI {
+  std::string scheme;
+  std::string authority;
+  std::string path;
+};
```

Comment:
This looks a bit like an ad-hoc utility masquerading as a generic one. URI has more parts - it's just that we're not necessarily interested in full parsing at present.
Perhaps a better way would be to store the offsets to the constituent parts and providing some inline utility functions that produce the interesting subsections.

```suggestion
struct ParsedURI {
  std::string URI;  // the original URI string
  std::ptrdiff_t authority_ofs, path_ofs, query_ofs, fragment_ofs;
  std::string_view authority() const;
  std::string_view path() const;
  std::string_view query() const;
  std::string_view path_and_query() const;  // many URI handling classes have this one
  std::string_view fragment() const;
};
```

Reply:
Done